### PR TITLE
Added `force_destroy` to `databricks_external_location` resource

### LINKED ISSUE #2352: [ISSUE] Issue with destroying `databricks_external_location` resource when a table has been created
It's not possible to destroy the  `databricks_external_location` resource after creating (and destroying) a table in that location without using force. However, the terraform provider doesn't have a `force_destroy` parameter.

If only the catalog and/or schema are created then destruction is possible without force.

### Configuration
```hcl
terraform {
  required_version = " >= 1.4.5"

  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "=4.50.0"
    }
    time = {
      source  = "hashicorp/time"
      version = "=0.9.1"
    }
    databricks = {
      source  = "databricks/databricks"
      version = "=1.17.0"
    }
  }
}
provider "aws" {
  region = "eu-west-1"
}

provider "databricks" {
  alias     = "workspace"
  auth_type = "basic"
  host      = ""
  username  = ""
  password  = ""
}

locals {
  role_name             = "databricks-self-assuming-role"
  bucket_name           = "my-uc-bucket-with-a-random-suffix"
  databricks_account_id = ""
  prefix                = "my-uc-external-path"
  catalog_name          = "my_catalog"
  schema_name           = "my_schema"
  cluster_id            = null
}

data "aws_caller_identity" "this" {}

data "aws_iam_policy_document" "assume" {
  statement {
    actions = [
      "sts:AssumeRole",
    ]
    principals {
      type = "AWS"
      identifiers = [
        "arn:aws:iam::${data.aws_caller_identity.this.account_id}:root",
        "arn:aws:iam::414351767826:role/unity-catalog-prod-UCMasterRole-14S5ZJVKOTYTL"
      ]
    }
    condition {
      test     = "StringEquals"
      values   = [local.databricks_account_id]
      variable = "sts:ExternalId"
    }
    condition {
      test = "ArnLike"
      values = [
        "arn:aws:iam::${data.aws_caller_identity.this.account_id}:role/${local.role_name}",
        "arn:aws:iam::414351767826:role/unity-catalog-prod-UCMasterRole-14S5ZJVKOTYTL",
      ]
      variable = "aws:PrincipalArn"
    }
  }
}

data "aws_iam_policy_document" "s3actions" {
  statement {
    actions = [
      "s3:GetObject",
      "s3:PutObject",
      "s3:DeleteObject",
      "s3:ListBucket",
      "s3:GetBucketLocation",
      "s3:GetLifecycleConfiguration",
      "s3:PutLifecycleConfiguration",
    ]
    resources = [
      "arn:aws:s3:::${aws_s3_bucket.this.bucket}/*",
      "arn:aws:s3:::${aws_s3_bucket.this.bucket}",
    ]
    effect = "Allow"
  }
  statement {
    actions = [
      "sts:AssumeRole",
    ]
    resources = [
      "arn:aws:iam::${data.aws_caller_identity.this.account_id}:role/${local.role_name}",
    ]
    effect = "Allow"
  }
}

resource "aws_iam_role" "this" {
  name               = local.role_name
  assume_role_policy = data.aws_iam_policy_document.assume.json
  inline_policy {
    name   = "resource-actions"
    policy = data.aws_iam_policy_document.s3actions.json
  }
}

resource "aws_s3_bucket" "this" {
  bucket        = local.bucket_name
  force_destroy = true # required because of the trail left by UC objects
}

# the prefix has to be in place otherwise the external location throws an error
resource "aws_s3_object" "object" {
  bucket = aws_s3_bucket.this.bucket
  key    = "${local.prefix}/"
}

resource "time_sleep" "wait_for_iam_role_propagation" {
  depends_on = [aws_iam_role.this]

  create_duration = "10s"
}

resource "databricks_storage_credential" "this" {
  provider = databricks.workspace
  name     = "my_storage_credentials"
  aws_iam_role {
    role_arn = aws_iam_role.this.arn
  }
  depends_on = [time_sleep.wait_for_iam_role_propagation]
}

resource "databricks_external_location" "this" {
  provider        = databricks.workspace
  credential_name = databricks_storage_credential.this.name
  name            = "my_external_location"
  url             = trimsuffix("s3://${aws_s3_object.object.bucket}/${aws_s3_object.object.key}", "/")
}

resource "databricks_catalog" "this" {
  provider     = databricks.workspace
  name         = local.catalog_name
  storage_root = "${databricks_external_location.this.url}/${local.catalog_name}"
}

resource "databricks_schema" "this" {
  provider     = databricks.workspace
  catalog_name = databricks_catalog.this.name
  name         = local.schema_name
  storage_root = "${databricks_external_location.this.url}/${local.catalog_name}/${local.schema_name}"
}

resource "databricks_sql_table" "this" {
  provider           = databricks.workspace
  catalog_name       = databricks_catalog.this.name
  schema_name        = databricks_schema.this.name
  table_type         = "MANAGED"
  data_source_format = "DELTA"
  name               = "my_test_table"
  cluster_id         = local.cluster_id
  column {
    name = "name"
    type = "string"
  }
}

```

### Expected Behavior
Everything should be destroyed including the external location.

### Actual Behavior
```
Error: cannot delete external location: Storage Location has dependent catalogs, schemas, managed or external tables/volumes; You may use force option to delete it but the managed storage data under this location cannot be purged by Unity Catalog anymore.
```

### Steps to Reproduce
1. `terraform apply`
2. `terraform destroy`

### Terraform and provider versions
```
Terraform v1.4.5
on darwin_arm64
+ provider registry.terraform.io/databricks/databricks v1.17.0
+ provider registry.terraform.io/hashicorp/aws v5.0.1
+ provider registry.terraform.io/hashicorp/time v0.9.1
```

### PR DESCRIPTION
**Pull request by ehab-eb**
## Changes
Fixes #2352 

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [ ] ~~using Go SDK~~

### CODE REVIEW

File: catalog/resource_external_location.go [hunk 1]
```diff
@@ -36,13 +36,20 @@ func (a ExternalLocationsAPI) get(name string) (el ExternalLocationInfo, err err
...
+	})
 }
 
 func ResourceExternalLocation() *schema.Resource {
 	s := common.StructToSchema(ExternalLocationInfo{},
 		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			m["force_destroy"] = &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
```
Comment (alexott) [inline catalog/resource_external_location.go:51]: this is not necessary, it's initialized to false anyway

**Review by alexott — CHANGES_REQUESTED** [review]
Please add acceptance test for this case.

**Comment by ehab-eb** [comment]
@alexott I've added a test but tbh I'm a little lost when trying to understand the wrapper functions. Does the wrapper destroy by default when it's done or should I set the argument to `true`? I tried checking other deletion tests for reference but didn't find a consistent pattern to follow. 
It's my first time using Go so apologies if the questions are a little basic :)

File: internal/acceptance/external_location_test.go [hunk 2]
```diff
@@ -31,3 +31,32 @@ func TestUcAccExternalLocation(t *testing.T) {
...
+		resource "databricks_storage_credential" "external" {
+			name = "cred-{var.RANDOM}"
+			aws_iam_role {
+				role_arn = "{env.TEST_METASTORE_DATA_ACCESS_ARN}"
+			}
+			comment = "Managed by TF"
+		}
+		
+		resource "databricks_external_location" "some" {
+			name            = "external"
```
Comment (nkvuong) [inline internal/acceptance/external_location_test.go:47]:
```suggestion
			name            = "external-{var.RANDOM}"
```

**Comment by nkvuong** [comment]
acceptance tests passed

**Review by nkvuong — APPROVED** [review]
lgtm

**Pull request merged**